IPython/core/application.py

```diff
@@ -195,6 +195,16 @@ def __init__(self, **kwargs):
     # Various stages of Application creation
     #-------------------------------------------------------------------------
 
+    def initialize_subcommand(self, subc, argv=None):
+        if subc in self.deprecated_subcommands:
+            import time
+            self.log.warning("Subcommand `ipython {sub}` is deprecated and will be removed "
+                             "in future versions.".format(sub=subc))
+            self.log.warning("You likely want to use `jupyter {sub}`... continue "
+                             "in 5 sec".format(sub=subc))
+            time.sleep(5)
```

Comment:
probably want to catch KeyboardInterrupt during the sleep to avoid traceback on abort.